### PR TITLE
Feature/52 build process

### PR DESCRIPTION
**Pull request by felix-zailskas**
Added a Dockefile to build and run the application easily and independent of the host machine. 

When run the application starts an interactive session that allows random fetching, prediction and data collection based on the previous demos. From now on the entrypoint of the application should be main.py (for future functionality).

Check README.md for instructions to run the application. Docker needs to be installed and running on your machine for this.

### CODE REVIEW

**Review by luccalb — APPROVED** [review]
Works fine!

**Review by ultiwinter — APPROVED** [review]
Worked on my side.

**Review by soapyheas — APPROVED** [review]
Run for me on my windows machine :). Looks good

File: src/main.py [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+    while True:
+        try:
+            choice = int(input("(1) BDC\n(2) EVP\n(3) DB\n(4) Pipeline\n"))
+            match choice:
+                case 1:
+                    bdc_demo()
+                case 2:
+                    evp_demo()
+                case 3:
+                    db_demo()
```
Comment (rbbozkurt) [inline src/main.py:16]: When db_demo() is selected before bdc_demo(), the app crashes. 
app_crush

**Review by rbbozkurt — APPROVED** [review]
Docker works fine on my side (macOS Monterey 12.6), but app crashes. 
Overall looks good. While I was at it, I thought I'd review it as well, and added some small notes for further development👍 .

File: src/database/database_dummy.py [hunk 2]
```diff
@@ -9,9 +9,9 @@
 
 
 class DatabaseDummy:
-    def __init__(self) -> None:
-        with open("src/database/dummy_leads.json") as f:
-            json_data = json.load(f)["training_leads"]
+    def __init__(self, source: str = "src/data/collected_data.json") -> None:
+        with open(source) as f:
```
[the remark this answers is not in the document]
Comment (felix-zailskas) [reply src/database/database_dummy.py]: Addressed

File: src/database/database_dummy.py [hunk 3]
```diff
@@ -20,6 +20,12 @@ def get_lead_by_id(self, id_: int) -> Lead:
     def get_all_leads(self) -> List[Lead]:
         return [LeadParser.parse_lead_from_dict(entry) for entry in self.data.values()]
 
+    def get_cardinality(self) -> int:
+        return len(self.data)
+
+    def get_cardinality(self) -> int:
```
[the remark this answers is not in the document]
Comment (felix-zailskas) [reply src/database/database_dummy.py]: Addressed

File: src/bdc/data_collector.py [hunk 4]
```diff
@@ -63,81 +68,15 @@ def get_data_from_api(self):
                         "company_address": users["company"]["address"]["address"],
                         "company_department": users["company"]["department"],
                         "company_name": users["company"]["name"],
+                        "annual_income": random.randint(0, 1000000),
```
[the remark this answers is not in the document]
Comment (felix-zailskas) [reply src/bdc/data_collector.py]: Addressed

File: src/main.py (hunk 1 above)
Comment (felix-zailskas) [reply src/main.py:16]: Addressed

**Review by ruchita-nathani — APPROVED** [review]
Working for me.

**Comment by ur-tech** [comment]
Everything is working fine!
I had some trouble because I used the Linux Subsystem on Windows and needed to adjust the format for my system (with dos2unix).

**Pull request merged**